pkg/controllers/edgeconnect/controller.go

```diff
@@ -132,38 +133,81 @@ func (controller *Controller) reconcileEdgeConnectDeletion(ctx context.Context,
 
 	edgeConnect.ObjectMeta.Finalizers = nil
 	if err := controller.client.Update(ctx, edgeConnect); err != nil {
-		_log.Debug("updating the EdgeConnect object failed")
+		_log.Debug("reconcile deletion: updating the EdgeConnect object failed")
 
 		return errors.WithStack(err)
 	}
 
 	edgeConnectClient, err := controller.buildEdgeConnectClient(ctx, edgeConnect)
 	if err != nil {
-		_log.Debug("building EdgeConnect client failed")
+		_log.Debug("reconcile deletion: building EdgeConnect client failed")
 
 		return err
 	}
 
 	tenantEdgeConnect, err := getEdgeConnectByName(edgeConnectClient, edgeConnect.Name)
 	if err != nil {
+		_log.Debug("reconcile deletion: failed to get EdgeConnect by name")
+
 		return err
 	}
 
 	switch {
 	case tenantEdgeConnect.ID == "":
-		_log.Info("EdgeConnect not found on the tenant")
+		{
+			_log.Info("EdgeConnect not found on the tenant")
+
+			return nil
+		}
 	case !tenantEdgeConnect.ManagedByDynatraceOperator:
-		_log.Info("can't delete EdgeConnect configuration from the tenant because it has been created manually by a user")
-	case edgeConnectIdFromSecret == "":
-		_log.Info("EdgeConnect client secret is missing")
+		{
+			_log.Info("can't delete EdgeConnect configuration from the tenant because it has been created manually by a user")
 
-		return edgeConnectClient.DeleteEdgeConnect(tenantEdgeConnect.ID)
+			return nil
+		}
+	case edgeConnectIdFromSecret == "":
+		{
+			_log.Info("EdgeConnect client secret is missing")
+		}
 	default:
-		if tenantEdgeConnect.ID != edgeConnectIdFromSecret {
-			_log.Info("EdgeConnect client secret contains invalid Id")
+		{
+			if tenantEdgeConnect.ID != edgeConnectIdFromSecret {
+				_log.Info("EdgeConnect client secret contains invalid Id")
+			}
 		}
```

Comment:
just for my understanding:

Why are we doing this?
- Extra cases for log lines, that do not do anything else?
- These log lines read more like debug logs
- At the end of the day, do we care about these in context of a deletion?

Like, the problem `"EdgeConnect client secret contains invalid Id"`, sounds like something that should never happen, and if it does then it should be handled and logged, BUT I (personally) don't care about it during deletion.

Reply:
This is a debugging session desperation leftover, to make sure that the switch statement wasn't just running the next code line and jumping the rest of the code :laughing: I will remove it :+1: